plugins/out_s3/s3.c

```diff
@@ -1075,7 +1099,24 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
         m_upload->upload_state = MULTIPART_UPLOAD_STATE_CREATED;
     }
 
-    ret = upload_part(ctx, m_upload, body, body_size);
+    if (chunk) {
+        create_time = chunk->create_time;
+    }
+    else {
+        create_time = time(NULL);
+    }
+    if(payload_size < MIN_CHUNKED_UPLOAD_SIZE) {
+        ret = s3_put_object(ctx, tag, create_time, payload_buf, payload_size);
+        flb_free(payload_buf);
+        if (ret >= 0) {
+            // m_upload->upload_state = 0;
+            flb_plg_info(ctx->ins, "Will complete upload for %s because One part of this chunk size is smaller than min upload size", m_upload->s3_key);
+            return FLB_OK;
+        }
+    } else {
+        ret = upload_part(ctx, m_upload, payload_buf, payload_size);
+        flb_free(payload_buf);
+    }
```

Comment:
As we discussed yesterday, I think the simplest way to do is to not change any of the code under the `put_object` and `multipart` labels, as long as the existing `body` and `body_size` variables are set to the correct final values... everything should still work..

Reply:
will change it